flatten_json.go

```diff
@@ -475,6 +496,60 @@ func (fj *flattenJSON) readLiteral(literal []byte) ([]byte, error) {
 	return literal, nil
 }
 
+func (fj *flattenJSON) skipBlock(openSymbol byte, closeSymbol byte) error {
+	level := 0
+
+	for fj.eventIndex < len(fj.event) {
+		ch := fj.event[fj.eventIndex]
+
+		switch ch {
+		case '"':
+			err := fj.skipStringValue()
+			if err != nil {
+				return err
+			}
+		case openSymbol:
+			level++
+		case closeSymbol:
+			level--
+
+			if level == 0 {
+				return nil
+			}
+		}
+
+		fj.eventIndex++
+	}
+
+	return fj.error("truncated block")
+}
+
+func (fj *flattenJSON) skipStringValue() error {
+	if fj.step() != nil {
+		return fj.error("event truncated in mid-string")
+	}
+
+	i := 0
+	data := fj.event[fj.eventIndex:]
+
+	for i < len(data) {
+		c := data[i]
+
+		if c == '\\' && i+1 < len(data) && data[i+1] == '"' {
```

Comment:
I'd put a short comment here explaining why you can skip all the other `\\` escapes, because the only thing you care about is an unescaped `"`

Reply:
Will add!